augur/argparse_.py

```diff
@@ -68,6 +68,10 @@ def add_command_subparsers(subparsers, commands, command_attribute='__command__'
         if not getattr(command, "run", None):
             add_default_command(subparser)
 
+        # Recursively register any subcommands
+        if getattr(subparser, "subcommands", None):
+            register_commands(subparser, subparser.subcommands)
```

Comment:
This breaks `augur curate` subcommands

```
$ cat records.ndjson | augur curate passthru
Traceback (most recent call last):
  File "…/augur/__init__.py", line 72, in run
    return args.__command__.run(args)
TypeError: run() missing 1 required positional argument: 'records'
```

because each subcommand's `run` function is expected to be called through [augur curate's own `run` function](https://github.com/nextstrain/augur/blob/710f8042bd8effb1240cf5e8c979e36de4c158cb/augur/curate/__init__.py#L140-L141), but it is now called directly at the [top level](https://github.com/nextstrain/augur/blob/710f8042bd8effb1240cf5e8c979e36de4c158cb/augur/__init__.py#L71).

Maybe recursively registering subcommands is not the right thing to do in this codebase.

Reply:
I remember having to work around the top level run when initially adding the `augur curate` subcommand. 

https://github.com/nextstrain/augur/blob/129fd7e9d36c38a94f33998fba8398ed92f73428/augur/curate/__init__.py#L117-L120

Would it be weird to keep `augur curate` as a special case?

Reply:
Two options:

1. Further adapt how `augur curate` commands share functionality so calling their `run()` from the top-level works. This could be done relatively straightforwardly.
2. Don't recursively register subcommands; leave the pattern of commands with subcommands invoking `register_commands(…)` themselves.

I don't think either gets the way of the functional improvements we want to make? So it's a matter of preference.

Reply:
Thanks for the input – I'll try option 1 above to align with Nextstrain CLI code.